tests/Server/TextDocumentTest.php

```diff
@@ -218,6 +218,16 @@ public function testFormatting()
             'newText' => $expected
         ]], json_decode(json_encode($result), true));
     }
+    
+    public function testFormattingInvalidUri()
+    {
+        $client =  new LanguageClient(new MockProtocolStream());
+        $project = new Project($client);
+        $textDocument = new Server\TextDocument($project, $client);
+        
+        $result = $textDocument->formatting(new TextDocumentIdentifier('whatever'), new FormattingOptions());
+        $this->assertEquals([], json_decode(json_encode($result), true));
```

Comment:
Actual implementation will create `PhpDocument` for invalid/non-existing uri (`Project::getDocument`). I can try to reorganize this with this change but I think it should be done as separate issue.